### PR TITLE
rich messaging support for all examples

### PR DESCRIPTION
**Pull request by spencerlepine**
### *Reference*:
- Announcement: https://aws.amazon.com/about-aws/whats-new/2022/03/amazon-connect-rich-chat-formatting/
- Documentation: https://docs.aws.amazon.com/connect/latest/adminguide/enable-text-formatting-chat.html
- S3 Deploy Script: https://github.com/spencerlepine/ui-examples-s3-deployment
- Chat-interface PR: https://github.com/amazon-connect/amazon-connect-chat-interface/pull/92

### Description of changes:

Update lambda code for rich messaging support for each example. Customer must enable the feature in `initiateChat()` config, and forward `supportedMessagingContentTypes` in `startChatContact` request.

Also [html formatter](https://marketplace.visualstudio.com/items?itemName=lonefy.vscode-JS-CSS-HTML-formatter) to clean up files.

> - [x] Update **`lambda`** code to pass `SupportedMessagingContentTypes` to ChatJS
> - [x] Update **`index.html`** initiateChat configs, enabling rich messaging
> - [x] Update **`READMEs`** with rich messaging support configuration
> - [x] Upgrade **`ACChat.js`** chat.js bundle file, for customChatWidget
> - [x] Update **`CHANGELOG`**

> **Note:** requires latest `ChatSDK.zip` using `aws-sdk@^v2.1295.0` (added in #87)

---

#### 🎯 Before Screenshot

old-chat-interface

#### 🎯 After Screenshot

chat_interface_rich_messaging

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by spencerlepine** [comment]
## Deployment

### [`startChatContactAPI`](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/startChatContactAPI)
 
| Region | S3 Updated | Stack Tested |
| ------ | -------- | ------- |
| us-east-1 (N. Virginia) |  <ul><li>[ ] </li></ul>  | <ul><li>[ ] </li></ul> |
| us-west-2 (Oregon) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-southeast-2 (Sydney) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-northeast-1 (Tokyo) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| eu-central-1 (Frankfurt) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| eu-west-2 (London) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-southeast-1 (Singapore) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ca-central-1 (Canada) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |

Deployed to s3 bucket in following regions:

- [ ] `ap-northeast-1.start-chat-contact-api-cfn` Asia Pacific (Tokyo) ap-northeast-1
- [ ] `ap-southeast-1.start-chat-contact-api-cfn` Asia Pacific (Singapore) ap-southeast-1
- [ ] `ap-southeast-2.start-chat-contact-api-cfn` Asia Pacific (Sydney) ap-southeast-2
- [ ] `ca-central-1.start-chat-contact-api-cfn` Canada (Central) ca-central-1
- [ ] `eu-central-1.start-chat-contact-api-cfn` EU (Frankfurt) eu-central-1	
- [ ] `eu-west-2.start-chat-contact-api-cfn` EU (London) eu-west-2	
- [ ] `us-east-1.start-chat-contact-api-cfn` US East (N. Virginia) us-east-1
- [ ] `us-west-2.start-chat-contact-api-cfn` US West (Oregon) us-west-2

---

### [`urlPreviewForAsyncChat`](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/urlPreviewForAsyncChat)
 
| Region | S3 Updated | Stack Tested |
| ------ | -------- | ------- |
| us-east-1 (N. Virginia) |  <ul><li>[ ] </li></ul>  | <ul><li>[ ] </li></ul> |
| us-west-2 (Oregon) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-southeast-2 (Sydney) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-northeast-1 (Tokyo) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| eu-central-1 (Frankfurt) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| eu-west-2 (London) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-southeast-1 (Singapore) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ca-central-1 (Canada) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |

Deployed to s3 bucket in following regions:

- [ ] `ap-northeast-1.amazon-connect-url-preview-for-async-chat-cfn` Asia Pacific (Tokyo) ap-northeast-1
- [ ] `ap-southeast-1.amazon-connect-url-preview-for-async-chat-cfn` Asia Pacific (Singapore) ap-southeast-1
- [ ] `ap-southeast-2.amazon-connect-url-preview-for-async-chat-cfn` Asia Pacific (Sydney) ap-southeast-2
- [ ] `ca-central-1.amazon-connect-url-preview-for-async-chat-cfn` Canada (Central) ca-central-1
- [ ] `eu-central-1.amazon-connect-url-preview-for-async-chat-cfn` EU (Frankfurt) eu-central-1	
- [ ] `eu-west-2.amazon-connect-url-preview-for-async-chat-cfn` EU (London) eu-west-2	
- [ ] `us-east-1.amazon-connect-url-preview-for-async-chat-cfn` US East (N. Virginia) us-east-1
- [ ] `us-west-2.amazon-connect-url-preview-for-async-chat-cfn` US West (Oregon) us-west-2

---

### [`asyncCustomerChatUX `](https://github.com/amazon-connect/amazon-connect-chat-ui-examples/tree/master/cloudformationTemplates/asyncCustomerChatUX)
 
| Region | S3 Updated | Stack Tested |
| ------ | -------- | ------- |
| us-east-1 (N. Virginia) |  <ul><li>[ ] </li></ul>  | <ul><li>[ ] </li></ul> |
| us-west-2 (Oregon) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-southeast-2 (Sydney) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-northeast-1 (Tokyo) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| eu-central-1 (Frankfurt) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| eu-west-2 (London) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ap-southeast-1 (Singapore) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |
| ca-central-1 (Canada) | <ul><li>[ ] </li></ul> | <ul><li>[ ] </li></ul> |

Deployed to s3 bucket in following regions:

- [ ] `ap-northeast-1.amazon-connect-advanced-customer-chat-cfnn` Asia Pacific (Tokyo) ap-northeast-1
- [ ] `ap-southeast-1.amazon-connect-advanced-customer-chat-cfn` Asia Pacific (Singapore) ap-southeast-1
- [ ] `ap-southeast-2.amazon-connect-advanced-customer-chat-cfn` Asia Pacific (Sydney) ap-southeast-2
- [ ] `ca-central-1.amazon-connect-advanced-customer-chat-cfn` Canada (Central) ca-central-1
- [ ] `eu-central-1.amazon-connect-advanced-customer-chat-cfn` EU (Frankfurt) eu-central-1	
- [ ] `eu-west-2.amazon-connect-advanced-customer-chat-cfn` EU (London) eu-west-2	
- [ ] `us-east-1.amazon-connect-advanced-customer-chat-cfn` US East (N. Virginia) us-east-1
- [ ] `us-west-2.amazon-connect-advanced-customer-chat-cfn` US West (Oregon) us-west-2

**Comment by spencerlepine** [comment]
- Created S3 bucket in us-west-2 
- Dropped `cloudformation.yml` into bucket
- Dropped files into `deployment/` folder:
   - [`custom-resource-helper.zip`](https://s3.ap-northeast-1.amazonaws.com/ap-northeast-1.start-chat-contact-api-cfn/deployment/custom-resource-helper.zip): `customResourceHelper.js` and `metricsHelper.js`, zipped from both in a folder: custom-resource-helper
   - [`start-chat-contact.zip`](https://start-chat-contact-api-cfn-spenlep.s3.us-west-2.amazonaws.com/deployment/start-chat-contact.js.zip): file startChatContact.js`, zipped, and renamed to `start-chat-contact.js.zip`)
   - [`ChatSDK.zip`](https://start-chat-contact-api-cfn-spenlep.s3.us-west-2.amazonaws.com/deployment/ChatSDK.zip): folder, with `node_modules/aws-sdk` nested, pre-packaged latest version of aws-cdk with Amazon.connect
- Created stack with button: [Launch Stack](https://us-west-2.console.aws.amazon.com/cloudformation/home#/stacks/new?stackName=startChatContactAPI&templateURL=https://start-chat-contact-api-cfn-spenlep.s3.us-west-2.amazonaws.com/cloudformation.yml)

**Comment by spencerlepine** [comment]
https://github.com/spencerlepine/ui-examples-s3-deployment

File: cloudformationTemplates/startChatContactAPI/customBuildIndex.html [hunk 1]
```diff
@@ -99,7 +96,9 @@ <h1>Amazon Connect - Custom Implementation of Customer Chat</h1>
                             DisplayName: customerName
                         },
                         ContactFlowId: contactFlowId,
-                        InstanceId: instanceId
+                        InstanceId: instanceId,
+                        // Enable rich messaging support: https://docs.aws.amazon.com/connect/latest/adminguide/enable-text-formatting-chat.html
+                        SupportedMessagingContentTypes: ["text/plain", "text/markdown"] // include 'text/markdown'
```
Comment (haomingli2020) [inline cloudformationTemplates/startChatContactAPI/customBuildIndex.html:101]: Why is this a list here but a string in other index.html files?

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,12 @@
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [1.3.1] - 2023-2-23
+### Added
+- Custom Chat Widget - Include Rich Messaging suppport
+- cloudformationTemplates - Include Rich Messaging suppport
+  - Amazon Connect Chat introduces rich messaging support TODO. [Learn more](https://docs.aws.amazon.com/connect/latest/adminguide/enable-text-formatting-chat.html)
```
[the remark this answers is not in the document]
Comment (spencerlepine) [reply CHANGELOG.md:10]: updating in next revision, missing full announcement

File: cloudformationTemplates/startChatContactAPI/customBuildIndex.html (hunk 1 above)
Comment (spencerlepine) [reply cloudformationTemplates/startChatContactAPI/customBuildIndex.html:101]: `connect.startChatContact()` and ChatInitializer will convert the string to the array, before sending to APIGW. This file calls APIGW directly, without going through `amazon-connect-chat-interface.js`

File: customChatWidget/public/amazon-connect-chat-interface.js.LICENSE.txt [hunk 3]
```diff
@@ -4,6 +4,13 @@ object-assign
 @license MIT
 */
 
+/*!
```
[the remark this answers is not in the document]
Comment (spencerlepine) [reply customChatWidget/public/amazon-connect-chat-interface.js.LICENSE.txt:7]: yes, `amazon-connect-chat-interface/build/dist/static/js/amazon-connect-chat-interface.js.LICENSE.txt`

File: cloudformationTemplates/startChatContactAPI/README.md [hunk 4]
```diff
@@ -347,6 +347,38 @@ If you want to enable sending attachments for Amazon Connect Chat the customer c
...
+    // ...
+    supportedMessagingContentTypes: "text/plain,text/markdown", // include 'text/markdown' for rich messaging support
+    featurePermissions: {
+      ATTACHMENTS: false,
+      MESSAGING_MARKDOWN: true
+    }
+  },successHandler, failureHandler)
+```
+
+2. Be sure to pass `supportedMessagingContentTypes` input to `startChatContact()`:
```
[the remark this answers is not in the document]
Comment (spencerlepine) [reply cloudformationTemplates/startChatContactAPI/README.md:369]: That is correct, adding note about existing CFN stack now. Existing solutions will throw 4xx errors, and be difficult to debug, so it is important to include this note here.

**Review by jiahaoyu6666 — APPROVED** [review]

**Review by haomingli2020 — APPROVED** [review]

**Pull request merged**